cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -177,47 +124,92 @@ protected final static String getToolWithEdition(String tool, String edition) {
+  public void run() {
+
+    runTool(this.arguments.asArray());
+  }
+
   /**
-   * @return the {@link EnvironmentVariables#getToolVersion(String) tool version}.
+   * @param args the command-line arguments to run the tool.
+   * @see ToolCommandlet#runTool(ProcessMode, GenericVersionRange, String...)
    */
-  public VersionIdentifier getConfiguredVersion() {
+  public void runTool(String... args) {
 
-    return this.context.getVariables().getToolVersion(getName());
+    runTool(ProcessMode.DEFAULT, null, args);
+  }
+
+  /**
+   * Ensures the tool is installed and then runs this tool with the given arguments.
+   *
+   * @param processMode the {@link ProcessMode}. Should typically be {@link ProcessMode#DEFAULT} or {@link ProcessMode#BACKGROUND}.
+   * @param toolVersion the explicit {@link GenericVersionRange version} to run. Typically {@code null} to run the
+   *     {@link #getConfiguredVersion() configured version}. Otherwise, the specified version will be used (from the software repository, if not compatible).
+   * @param args the command-line arguments to run the tool.
+   */
+  public final void runTool(ProcessMode processMode, GenericVersionRange toolVersion, String... args) {
+
+    runTool(processMode, toolVersion, ProcessErrorHandling.THROW_CLI, args);
   }
 
   /**
-   * Method to be called for {@link #install(boolean)} from dependent {@link com.devonfw.tools.ide.commandlet.Commandlet}s. Additionally, contains the
-   * environmentContext of the tool
+   * Ensures the tool is installed and then runs this tool with the given arguments.
    *
-   * @param environmentContext - the environment context that can be used for the dependencies
-   * @return {@code true} if the tool was newly installed, {@code false} if the tool was already installed before and nothing has changed.
+   * @param processMode the {@link ProcessMode}. Should typically be {@link ProcessMode#DEFAULT} or {@link ProcessMode#BACKGROUND}.
+   * @param toolVersion the explicit {@link GenericVersionRange version} to run. Typically {@code null} to run the
+   *     {@link #getConfiguredVersion() configured version}. Otherwise, the specified version will be used (from the software repository, if not compatible).
+   * @param args the command-line arguments to run the tool.
+   */
+  public ProcessResult runTool(ProcessMode processMode, GenericVersionRange toolVersion, ProcessErrorHandling errorHandling, String... args) {
+
+    ProcessContext pc = this.context.newProcess().errorHandling(errorHandling);
+    install(true, pc);
+    configureToolBinary(pc, processMode, errorHandling);
+    configureToolArgs(pc, processMode, errorHandling, args);
+    return pc.run(processMode);
+  }
+
+  /**
+   * @param pc the {@link ProcessContext}.
+   * @param processMode the {@link ProcessMode}.
+   * @param errorHandling the {@link ProcessErrorHandling}.
+   */
+  protected void configureToolBinary(ProcessContext pc, ProcessMode processMode, ProcessErrorHandling errorHandling) {
```

Comment:
Why was errorHandling param added and not used?

Reply:
Same answer as your analoge questions below, but currently we do not use and override it. I just like to think and plan ahead and already know that it will be used later :)